test/controller_test.dart

```diff
@@ -34,6 +50,10 @@ void main() {
     await tester.tap(find.byKey(snackBar));
     await tester.pump();
     expect(find.text('Hi'), findsOneWidget);
+
+    expect(numberOfWidgetBuilds, equals(1));
```

Comment:
@ShadyBoukhary so, to perform your request, I added some methods inside the page that will increment variables each time `build` is called inside every widget.

If you check the test, it correctly validates that both parent builder and uncontrolled widget builder runs only one time, and just the controlled widget runs three times (the number of times that the variable on controller changes)

Reply:
Sounds good!